cibuild.sh

```diff
@@ -0,0 +1,171 @@
+             cd $prebuilt; \
+             wget https://static.dev.sifive.com/dev-tools/riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-linux-ubuntu14.tar.gz; \
+             tar zxf riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-linux-ubuntu14.tar.gz; \
+             mv riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-linux-ubuntu14 riscv64-unknown-elf-gcc; \
+             rm riscv64-unknown-elf-gcc-8.3.0-2019.08.0-x86_64-linux-ubuntu14.tar.gz; \
+           fi; \
+           riscv64-unknown-elf-gcc --version; \
+           ")
+path+=("$prebuilt/riscv64-unknown-elf-gcc/bin")
+
+# ccache
+install+=("if [ ! -d \"$prebuilt/ccache/bin\" ]; then \
+             cd $prebuilt; \
+             wget https://github.com/ccache/ccache/releases/download/v3.7.7/ccache-3.7.7.tar.gz; \
+             tar zxf ccache-3.7.7.tar.gz; \
+             cd ccache-3.7.7; ./configure --prefix=$prebuilt/ccache; make; make install; \
+             cd $prebuilt; rm -rf ccache-3.7.7; rm ccache-3.7.7.tar.gz; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/gcc; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/g++; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/arm-none-eabi-gcc; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/arm-none-eabi-g++; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/p32-gcc; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/riscv64-unknown-elf-gcc; \
+             ln -sf $prebuilt/ccache/bin/ccache $prebuilt/ccache/bin/riscv64-unknown-elf-g++; \
+           fi; \
+           ccache --version; \
+           export USE_CCACHE=1; \
+           export CCACHE_DIR=$prebuilt/ccache/.ccache; \
+           ccache -M 5G; \
+           ")
+path+=("$prebuilt/ccache/bin")
+
+function usage {
+  echo ""
+  echo "USAGE: $0 check | full"
+  echo "       $0 -h"
+  echo ""
+  echo "Where:"
+  echo "  check:   do check build CI"
+  echo "  full:    do full build CI"
+  echo "  -h will show this help test and terminate"
+  echo ""
+  exit 1
+}
+
+function setup_repos {
+  if [ -d "$nuttx" ]; then
+    cd $nuttx; git pull
+  else
+    git clone https://github.com/apache/incubator-nuttx.git $nuttx
+  fi
+
+  if [ -d "$apps" ]; then
+    cd $apps; git pull
+  else
+    git clone https://github.com/apache/incubator-nuttx-apps.git $apps
+  fi
+}
+
+function install_tools {
+  for ((i=0; i<${#install[@]}; ++i)); do
+    export PATH=${path[$i]}:$PATH
+    eval ${install[$i]}
+  done
+}
+
+function run_builds {
+  local JOBS=`grep -c ^processor /proc/cpuinfo`
+
+  $nuttx/tools/testbuild.sh -si -j $JOBS $WD/testlist/${build}list.dat
+  if [ $? != 0 ]; then
+    echo "ERROR: $BUILD build failed."
+    exit 1
+  fi
+}
+
+if [ -z "$1" ]; then
+   usage
+fi
+
+while [ ! -z "$1" ]; do
+  case "$1" in
+  -h )
+    usage
+    ;;
+  check | full )
+    build=$1
+    break
+    ;;
+  * )
```

Comment:
maybe having a additional argument to just install the tools could be useful, like `cibuild.sh install_tools`. This way a Docker image can be prepared with all the tools ready to go, and just start with an empty workspace (in benefit of speed with reasonable repeatability). In fact, a companion `Dockerfile` would be a nice extra (but it could wait for another pull-request).

Reply:
It's a good idea to provide install_tools for docker image. I'll update later.

Reply:
Thanks

Reply:
Besides added -i option for install_tools, -s option for setup_repos and -b for run_builds also added. I think it would also be good for Jenkins pipeline job.

Reply:
Good idea!